tensorflow/core/kernels/string_split_op.cc

```diff
@@ -28,19 +28,62 @@ namespace tensorflow {
 
 namespace {
 
-std::vector<string> Split(const string& str, const string& delimiter,
-                          const bool skipEmpty) {
+Status Split(std::vector<string>& char_vector, const string& str,
+             const string& delimiter, const bool skipEmpty,
+             const string& encoding) {
   if (!delimiter.empty()) {
     if (skipEmpty) {
-      return str_util::Split(str, delimiter, str_util::SkipEmpty());
+      char_vector = str_util::Split(str, delimiter, str_util::SkipEmpty());
+      return Status::OK();
     }
-    return str_util::Split(str, delimiter);
+    char_vector = str_util::Split(str, delimiter);
+    return Status::OK();
   }
-  std::vector<string> char_vector(str.size());
-  for (size_t i = 0; i < str.size(); ++i) {
-    char_vector[i] = str[i];
+  if (encoding == "utf8") {
+    // Bytes    Byte 1    Byte 2    Byte 3    Byte 4
```

Comment:
Could we push this upstream to str_util and provide a variant there? It might be useful for the rest of the code. If moved there, it will also have to have a unit test on its own.

Reply:
Done.